### DISCUSSION TITLE
Very high false positive results

### DISCUSSION THREAD

**Discussion by tiamens** [exadel-inc/CompreFace#560] [Q&A]
im been testing compreface a while for door lock project. i dont know why but my results are not very good at the moment. I have trained the faces of 4 people to the system so far.  a lot of time compreface give me a false positive with %95 accuracy. Sometimes it recognizes the filament box as a human face and claims to be one of the faces I trained.

i use ip cameras with full hd resolution. These cameras new and have a decent optics in it. 

I've looked through the documentation, but I haven't come across a setting that can fix this situation. i trained faces whit high resolution and specifically cropped for better training.

im doing something wrong maybe i dont know but thats very disappointing situation for me right know.  Thanks to the exadel team for great solution. i hope it will better in the future.

**Comment by IvanKurnosov** [comment]
Hello! The problem could be on a different stages of face recognition process. Could you send 4 photos, that you add to compreface system and a few other photos, where compreface gives you a false positive answer (like a filament box you mentioned). First i'll run our embedding calculator on this photos and calculate the similarity manually to check our models. Then i'll run our entire service on this photos to see it behavior.

You may sand me photos here or to my email: ikurnosau@exadel.com, if you don't want to share them pyblicly.

**Reply by tiamens** [reply]
Ivan hi, i will sent you the photos (same person) and a image from false positive match. i hope it help. 

thank you for wonderful solution again.

**Comment by nxliu** [comment]
we face the same problem,  and make compreface a non-valid solution！ we use our team member for testing ,  and will return high false positive match 98% from member A with member B, any suggestion?

**Reply by pospielov** [reply]
Hi, could you send us examples of the photos?
compreface.support@exadel.com
Also, you can try a custom build, SubCenter-ArcFace-r100 build should be the most accurate:
https://github.com/exadel-inc/CompreFace/blob/master/custom-builds/README.md

**Reply by nxliu** [reply]
thanks a lot, i have send the photos to the above email,  any suggestion is welcome, we will try the custom build

**Reply by martinenkoEduard** [reply]
Btw do you have any readmes, on how I can additionally teach this mode?

**Reply by pospielov** [reply]
We don't, but you can find instructions in the InsightFace library:
https://github.com/deepinsight/insightface/tree/master/recognition/arcface_mxnet
